crates/lair_keystore/src/store_sqlite.rs

```diff
@@ -163,18 +163,35 @@ impl SqlPool {
         // initialize the sqlcipher key pragma
         let key_pragma = secure_write_key_pragma(dbk_secret)?;
 
-        // open a single write connection to the database
-        let mut write_con = rusqlite::Connection::open_with_flags(
-            &path,
-            OpenFlags::SQLITE_OPEN_READ_WRITE
-                | OpenFlags::SQLITE_OPEN_CREATE
-                | OpenFlags::SQLITE_OPEN_NO_MUTEX
-                | OpenFlags::SQLITE_OPEN_URI,
-        )
-        .map_err(one_err::OneErr::new)?;
-
-        // set generic pragmas
-        set_pragmas(&write_con, key_pragma.clone())?;
+        let mut write_con =
+            match create_configured_db_connection(&path, key_pragma.clone()) {
+                Ok(con) => con,
+                Err(e) => {
+                    if "true"
+                        == std::env::var("LAIR_MIGRATE_UNENCRYPTED")
+                            .unwrap_or_default()
+                            .as_str()
+                    {
+                        // Known SQLite error when the database is not encrypted, try to encrypt it and retry starting the server
+                        if let Some("file is not a database") =
```

Comment:
Could improve this by changing the connection function to not return a `one_err::OneErr` and match the real type. Then convert if it doesn't match.

Reply:
Much better